src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -148,45 +148,96 @@ private void DoWorkExpectCancel(object sender, DoWorkEventArgs e)
             manualResetEvent3.Set();
         }
 
+        #endregion
+
         [Fact]
         public void TestThrowExceptionInDoWork()
         {
-            var bw = new TestWorker();
-            bw.DoWork += DoWorkWithException;
-            bw.RunWorkerAsync("Exception");
+            const string expectedArgument = "Exception";
+            const string expectedExceptionMsg = "Exception from DoWork";
+            var bw = new BackgroundWorker();
+
+            bw.DoWork += (sender, e) =>
+            {
+                Assert.Same(bw, sender);
+                Assert.Same(expectedArgument, e.Argument);
+                throw new TestException(expectedExceptionMsg);
+            };
+
+            bw.RunWorkerCompleted += (sender, e) =>
+            {
+                TestException ex = Assert.Throws<TestException>(() => e.Result);
+                Assert.Equal(expectedExceptionMsg, ex.Message);
+            };
+
+            bw.RunWorkerAsync(expectedArgument);
```

Comment:
Yes, please. I didn't see the original test, but if we don't, the test could theoretically not finish by the time the tests were shutdown.  It's possible xunit handles this by hooking some additional members in SynchronizationContext, but even if it does, failures would be difficult to debug.